### PR TITLE
Add Forum option for Task type

### PR DESCRIPTION
**Pull request by dedywahyudi**
Add Forum option for Task type as screenshot flow below: 

Screen Shot 2021-09-21 at 12 06 48

Screen Shot 2021-09-21 at 12 25 37

Screen Shot 2021-09-21 at 12 41 32

### CODE REVIEW

File: src/components/ChallengeEditor/ChallengeViewTabs/index.js [hunk 1]
```diff
@@ -85,17 +85,17 @@ const ChallengeViewTabs = ({
       <div className={styles.topContainer}>
         <div className={styles.leftContainer}>
           <div className={styles.title}>{challenge.name}</div>
-          {!isTask && (
```
Comment (vikasrohit) [inline src/components/ChallengeEditor/ChallengeViewTabs/index.js:88]: @dedywahyudi I don't think we need to show legacy links to all types. This change would make legacy links visible to every challenge type.

File: src/components/ChallengeEditor/index.js [hunk 2]
```diff
@@ -135,6 +137,9 @@ class ChallengeEditor extends Component {
     this.updateResource = this.updateResource.bind(this)
     this.onDeleteChallenge = this.onDeleteChallenge.bind(this)
     this.deleteModalLaunch = this.deleteModalLaunch.bind(this)
+    this.toggleForumOnCreate = this.toggleForumOnCreate.bind(this)
+    this.onSelectForum = this.onSelectForum.bind(this)
+    this.toggleForumOnCreate = this.toggleForumOnCreate.bind(this)
```
Comment (vikasrohit) [inline src/components/ChallengeEditor/index.js:142]: This line is duplicate of 140

File: src/components/ChallengeEditor/index.js [hunk 3]
```diff
@@ -913,9 +929,18 @@ class ChallengeEditor extends Component {
...
+      if (this.state.hasForum) {
+        const discussions = this.getDiscussionsConfig(newChallenge)
+        if (discussions) {
+          newChallenge.discussions = discussions
+        }
+      }
+    } else {
+      const discussions = this.getDiscussionsConfig(newChallenge)
+      if (discussions) {
+        newChallenge.discussions = discussions
```
Comment (vikasrohit) [inline src/components/ChallengeEditor/index.js:942]: Code duplicity can be avoided here.

File: src/components/ChallengeEditor/index.js [hunk 4]
```diff
@@ -1435,6 +1461,7 @@ class ChallengeEditor extends Component {
         )
       }
     </React.Fragment>
+    const useTask = _.find(metadata.challengeTypes, { id: challenge.typeId, isTask: true })
```
Comment (vikasrohit) [inline src/components/ChallengeEditor/index.js:1464]: `isTask` variable is already there to determine if the challenge is actually a task.

File: src/components/ChallengeEditor/Forum-Field/index.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Select from '../../Select'
+import cn from 'classnames'
+import styles from './Forum-Field.module.scss'
+
+const ForumField = ({ types, onSelectForum, challenge, hasForum, disabled }) => {
```
Comment (vikasrohit) [inline src/components/ChallengeEditor/Forum-Field/index.js:7]: I don't see using ForumField anywhere. I think you are using Discussion-Field instead.

File: src/components/ChallengeEditor/Forum-Field/Forum-Field.module.scss [hunk 6]
```diff
@@ -0,0 +1,55 @@
+@import "../../../styles/includes";
```
Comment (vikasrohit) [inline src/components/ChallengeEditor/Forum-Field/Forum-Field.module.scss:1]: We don't need this file if ForumField component is not needed.

**Review by vikasrohit — CHANGES_REQUESTED** [review]
There are many improvements required here.

**Comment by dedywahyudi** [comment]
@vikasrohit 

Please check the updated code. 
Let me know if still need more edits.

File: src/components/ChallengeEditor/index.js [hunk 7]
```diff
@@ -1435,6 +1448,7 @@ class ChallengeEditor extends Component {
         )
       }
     </React.Fragment>
+    // const useTask = _.find(metadata.challengeTypes, { id: challenge.typeId, isTask: true })
```
Comment (vikasrohit) [inline src/components/ChallengeEditor/index.js:1451]: Please removed it from code.

File: src/components/ChallengeEditor/ChallengeViewTabs/index.js [hunk 8]
```diff
@@ -85,17 +86,19 @@ const ChallengeViewTabs = ({
...
-          )}
+
+          <div
+            className={cn(
+              styles.actionButtons,
+              styles.button,
+              styles.actionButtonsLeft
+            )}
+          >
+            { isTask ? (<ForumLink challenge={challenge} />)
```
Comment (vikasrohit) [inline src/components/ChallengeEditor/ChallengeViewTabs/index.js:97]: @dedywahyudi this logic would not render the forum link for non task challenges, is it what intended here?

Comment (dedywahyudi) [reply src/components/ChallengeEditor/ChallengeViewTabs/index.js:97]: @vikasrohit 
{ isTask ? (<ForumLink challenge={challenge} />) : (<LegacyLinks challenge={challenge} challengeView />) }

So, Non Task challenge render by LegacyLinks component already. But, forum link use new ForumLink component

File: src/components/ChallengeEditor/index.js (hunk 7 above)
Comment (dedywahyudi) [reply src/components/ChallengeEditor/index.js:1451]: @vikasrohit it look we need this const useTask, somehow it not working if just use isTask.

Comment (dedywahyudi) [reply src/components/ChallengeEditor/index.js:1451]: in Create New Work screen isTask is always false , so that we have to use another const useTask

**Pull request merged**